plugins/geojson/src/main/java/com/mapbox/mapboxsdk/plugins/geojson/GeoJsonPlugin.java

```diff
@@ -0,0 +1,441 @@
+        }
+
+        @Override
+        protected void onPostExecute(DataModel dataModel) {
+            super.onPostExecute(dataModel);
+            DrawOnMap(dataModel);
+            if (loadURL != null) {
+                loadURL.onLoaded();
+            }
+        }
+    }
+
+    /**
+     * AsyncTask that loading GeoJson file from special path
+     */
+    private class DrawGeoJsonFromPath extends AsyncTask<Void, Void, DataModel> {
+        @Override
+        protected void onPreExecute() {
+            if (loadFilePath != null)
+                loadFilePath.onPreLoading();
+        }
+
+        @Override
+        protected DataModel doInBackground(Void... voids) {
+            DataModel dataModel = null;
+            try {
+                InputStream inputStream = new FileInputStream(filePath);
+                dataModel = parseGeoJsonInputStream(inputStream, loadFilePath);
+            } catch (FileNotFoundException e) {
+                loadFilePath.onLoadFailed(e);
+            }
+            return dataModel;
+        }
+
+        @Override
+        protected void onPostExecute(DataModel dataModel) {
+            super.onPostExecute(dataModel);
+            if (dataModel != null) {
+                DrawOnMap(dataModel);
+                if (loadFilePath != null)
+                    loadFilePath.onLoaded();
+            }
+        }
+    }
+
+    /**
+     * AsyncTask that loading GeoJson file from Assets folder
+     */
+    private class DrawGeoJsonFromAssets extends AsyncTask<Void, Void, DataModel> {
+        @Override
+        protected void onPreExecute() {
+            if (loadFileAssets != null)
+                loadFileAssets.onPreLoading();
+        }
+
+        @Override
+        protected DataModel doInBackground(Void... voids) {
+            DataModel dataModel = null;
+            try {
+                InputStream inputStream = context.getAssets().open(fileName);
+                dataModel = parseGeoJsonInputStream(inputStream, loadFileAssets);
+            } catch (IOException e) {
+                loadFileAssets.onLoadFailed(e);
+            }
+
+            return dataModel;
+        }
+
+        @Override
+        protected void onPostExecute(DataModel dataModel) {
+            super.onPostExecute(dataModel);
+            if (dataModel != null) {
+                DrawOnMap(dataModel);
+                if (loadFileAssets != null)
+                    loadFileAssets.onLoaded();
+            }
+        }
+    }
+
+    /**
+     * @param geoJson String of GeoJson file
+     * @return DataModel list of polylines, polygons and point with bounded
+     */
+    private DataModel parseGeoJsonString(String geoJson) {
+        int pointCount = 0;
+        DataModel dataModel = new DataModel();
+        LatLngBounds.Builder builder = new LatLngBounds.Builder();
+        try {
+            JSONObject json = new JSONObject(geoJson);
+            JSONArray features = json.getJSONArray("features");
```

Comment:
We have an [open source library](https://www.mapbox.com/android-docs/api/mapbox-java/libjava-geojson/2.1.0/index.html) that handles GeoJson transparently without the need to use the lower level `JSONObject` classes. Any reason not to use for the plugin? It comes includes by default with the Maps SDK so you should be able to use it without having to add any external dependencies.